src/entitysdk/core.py

```diff
@@ -86,6 +88,38 @@ def get_entity(
     return serdes.deserialize_model(response.json(), entity_type)
 
 
+def get_entity_derivations(
+    *,
+    api_url: str,
+    entity_id: ID,
+    entity_type: type[Entity],
+    project_context: ProjectContext,
+    derivation_type: DerivationType | None,
+    token: str,
+    http_client: httpx.Client | None = None,
+):
+    """Get derivations for entity."""
+    url = get_entity_derivations_endpoint(
+        api_url=api_url,
+        entity_type=entity_type,
+        entity_id=entity_id,
+    )
+
+    params = {"derivation_type": DerivationType(derivation_type)} if derivation_type else None
```

Comment:
I don't see `derivation_type` in the endpoint parameters.
We can add them if needed, but in that case it should be made clear how to select the derivations with derivation_type=None, and how to select all the derivations regardless of the type.

Reply:
Yeah, I added this because we want to be able to do that eventually, but you are right that it won't work for now.

Reply:
> I would suggest that if a derivation_type is not provided then all are fetched, including None

This makes sense, and how to select only derivation_type=None? However, we can think about it later, and we can also decide to migrate None to a specific value, and forbid null values.

Reply:
Yeah, I think we may need to specify all the allowed derivations per entity similar to how we do with asset labels and disallow nulls.